cwltool/job.py

```diff
@@ -394,25 +407,30 @@ def run(self, pull_image=True, rm_container=True,
         if self.generatemapper:
             self.add_volumes(self.generatemapper, runtime)
 
+        if user_space_docker_cmd:
+            runtime = [x.replace(":ro", "") for x in runtime]
+            runtime = [x.replace(":rw", "") for x in runtime]
+
         runtime.append(u"--workdir=%s" % (docker_windows_path_adjust(self.builder.outdir)))
+        if not user_space_docker_cmd:
 
-        if not kwargs.get("no_read_only"):
-            runtime.append(u"--read-only=true")
+            if not kwargs.get("no_read_only"):
+               runtime.append(u"--read-only=true")
 
-        if kwargs.get("custom_net", None) is not None:
-            runtime.append(u"--net={0}".format(kwargs.get("custom_net")))
-        elif kwargs.get("disable_net", None):
-            runtime.append(u"--net=none")
+            if kwargs.get("custom_net", None) is not None:
+                runtime.append(u"--net={0}".format(kwargs.get("custom_net")))
+            elif kwargs.get("disable_net", None):
+                runtime.append(u"--net=none")
 
-        if self.stdout:
-            runtime.append("--log-driver=none")
+            if self.stdout:
+                runtime.append("--log-driver=none")
 
-        euid, egid = docker_vm_id()
-        if not onWindows():  # MS Windows does not have getuid() or geteuid() functions
-            euid, egid = euid or os.geteuid(), egid or os.getgid()
+            euid, egid = docker_vm_id()
+            if not onWindows():  # MS Windows does not have getuid() or geteuid() functions
+                euid, egid = euid or os.geteuid(), egid or os.getgid()
 
-        if kwargs.get("no_match_user", None) is False and (euid, egid) != (None, None):
-            runtime.append(u"--user=%d:%d" % (euid, egid))
+            if kwargs.get("no_match_user", None) is False and (euid, egid) != (None, None):
+                runtime.append(u"--user=%d:%d" % (euid, egid))
```

Comment:
`udocker` does support `--user`; can `dx-docker` just ignore this?

Reply:
Yes, good point. I'll add a feature to it to ignore it for now and add support down the line.  Here's the PR I have for ignoring: https://github.com/dnanexus/dx-toolkit/pull/254